modules/recon/domains-hosts/binaryedge_hosts.py

```diff
@@ -0,0 +1,24 @@
+from recon.core.module import BaseModule
+
+class Module(BaseModule):
+
+    meta = {
+        'name': 'BinaryEdge.io DNS lookup',
+        'author': 'Ryan Hays',
+        'version': '1.0',
+        'description': 'Uses the BinaryEdge API to discover subdomains.',
+        'required_keys': ['binaryedge_api'],
+        'query': 'SELECT DISTINCT domain FROM domains WHERE domain IS NOT NULL',
+    }
+
+    def module_run(self, domains):
+        key = self.get_key('binaryedge_api')
+        for domain in domains:
+            self.heading(domain, level=0)
+            resp = self.request(url=f"https://api.binaryedge.io/v2/query/domains/dns/{domain}", headers={'X-Key': key})
+            if resp.status_code == 200:
+                for subdomain in resp.json().get('events'):
+                    try:
+                        self.insert_hosts(host=subdomain['domain'], ip_address=subdomain['A'][0])
+                    except KeyError:
+                        self.insert_hosts(host=subdomain['domain'])
```

Comment:
Looks pretty good. Short and sweet. But rather than use a try/except block, just use an `if` block to determine whether or not the ip address should be added. It's cleaner. Also, please add a newline to the end of the file. Thanks!

Reply:
Updated this